File: Programing/HigherOrderFunction/Exercise1/AnswerJava/Exercise1.java
import java.util.*;
import java.util.stream.*;

class Exercise1 {
    public void execute() {
	ExecutionUtils.ExecuteWithTimeMeasuring(this::sendFile);
	ExecutionUtils.ExecuteWithTimeMeasuring(this::sendBigSizeFile);
    }

    private void sendFile() {
	LinkedList<Integer> list = new LinkedList<Integer>();
	for (int i = 0; i < 1000000; i++) {
	    list.add(i);
	}
	System.out.println(list.size());
    }

    private void sendBigSizeFile() {
	LinkedList<Integer> list = new LinkedList<Integer>();
	for (int i = 0; i < 6000000; i++) {
	    list.add(i);
	}
	System.out.println(list.size());
    }
}

File: Programing/HigherOrderFunction/Exercise1/AnswerCSharp/Exercise1.cs
using System;
using System.Collections.Generic;

namespace Exercise1
{
    class Exercise1
    {
        public void Execute()
	{
	    ExecutionUtils.ExecuteWithTimeMeasuring(sendFile);
	    ExecutionUtils.ExecuteWithTimeMeasuring(sendBigSizeFile);
	}

        private void sendFile()
	{
	    var list = new List<int>();
	    for (int i = 0; i < 1000000; i++)
	    {
	        list.Add(i);
	    }
	    Console.WriteLine(list.Count);
	}

        private void sendBigSizeFile()
	{
	    var list = new List<int>();
	    for (int i = 0; i < 6000000; i++)
	    {
	        list.Add(i);
	    }
	    Console.WriteLine(list.Count);
	}
    }
}